src/acl/agentClass.test.ts

```diff
@@ -0,0 +1,546 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { describe, it, expect } from "@jest/globals";
+import { DataFactory } from "n3";
+import { dataset } from "@rdfjs/dataset";
+import {
+  unstable_getPublicResourceAccessModes,
+  unstable_getPublicDefaultAccessModes,
+  unstable_getPublicAccessModes,
+} from "./agentClass";
+import {
+  LitDataset,
+  WithResourceInfo,
+  IriString,
+  unstable_AccessModes,
+  unstable_AclDataset,
+  unstable_WithAcl,
+} from "../interfaces";
+
+function addAclRuleQuads(
+  aclDataset: LitDataset & WithResourceInfo,
+  resource: IriString,
+  accessModes: unstable_AccessModes,
+  type: "resource" | "default",
+  agentClass:
+    | "http://xmlns.com/foaf/0.1/Agent"
+    | "http://www.w3.org/ns/auth/acl#AuthenticatedAgent"
+): unstable_AclDataset {
+  const subjectIri =
+    resource + "#" + encodeURIComponent(agentClass) + Math.random();
+  aclDataset.add(
+    DataFactory.quad(
+      DataFactory.namedNode(subjectIri),
+      DataFactory.namedNode("http://www.w3.org/1999/02/22-rdf-syntax-ns#type"),
```

Comment:
This could use the `rdf.type` constant defined in `constants.ts`, but it's really minor. This also applies to:
- http://www.w3.org/ns/auth/acl#Authorization
- http://www.w3.org/ns/auth/acl#accessTo
- http://www.w3.org/ns/auth/acl#mode
- foaf:Agent

Reply:
Yep, but I've refrained from using conveniences from other parts of the codebase in the tests. It might be a bit overzealous in this case, but in case those constants were wrong, this means that the unit tests would catch that. We can change that, but then I suppose it's best to be consistent and do that over the entire code base (because all tests use exact URLs atm), i.e. in a separate PR :)

Reply:
Oh yeah right my bad, you told me already that you intentionnally kept the tests self-sufficient, that actually makes a lot of sense.